DEVELOPING.md

```diff
@@ -0,0 +1,137 @@
+# Builder Services Development Environment
+
+## Overview
+
+This document captures the steps to start and run a Builder environment for development. The builder environment includes the builder services, as well as the depot web site.
+
+There are several ways of creating a Builder dev environment - but supporting all operating systems and environments has proven to be untenable. This document includes one officially supported way of creating a Builder dev environment, and links to unsupported ways of creating the dev environment that you may use at your own risk.
+
+## Pre-Requisites
+
+### Ports required
+1. 9636 - Intra-supervisor communication
+1. 80 - Web app
+1. 9631 - supervisor api port
+1. 5432 - posgres port
+
+### Checkout
+1. If you are developing on Linux
+1. 1. Ensure you have curl
+1. 1. `git clone https://github.com/habitat-sh/builder.git /src/builder`
+1. If you are developing on a Mac, `git clone https://github.com/habitat-sh/builder.git`
+
+If you are using Linux environment you can run `/src/support/linux/provision.sh` to configure your host
+If you are on a Mac, you will need brew, direnv, habitat, and Docker for Mac
+
+### GitHub OAuth Application
+`APP_HOSTNAME` mentioned below, will typically be `localhost`
```

Comment:
Do we support this being anything _other_ than `localhost`? Based on our scripts, we don't so, maybe we shouldn't imply there's any flexibility here and say it _must_ be `localhost`. Or, we could figure out the right way to support other values.

Reply:
This can also be an IP if you happen to be running this on a local network or EC2 VM. Which scripts are you referring to that are hard-coded to something?

Reply:
[`origin`](https://github.com/habitat-sh/builder/pull/5/files#diff-240ab41eb291d125c5844f2ba10aa78fR294), [`project`](https://github.com/habitat-sh/builder/pull/5/files#diff-240ab41eb291d125c5844f2ba10aa78fR312), possibly others.

Reply:
ah, those will always be running locally but they are all localized to the studio. The times that we would need to care about external urls are for mostly the web stuff, which should be all encapsulated in `APP_HOSTNAME`